dbms/src/Storages/KVStore/MultiRaft/PrehandleSnapshot.cpp

```diff
@@ -42,6 +42,96 @@ extern const int TABLE_IS_DROPPED;
 extern const int REGION_DATA_SCHEMA_UPDATED;
 } // namespace ErrorCodes
 
+enum class ReadFromStreamError
+{
+    Ok,
+    Aborted,
+    ErrUpdateSchema,
+    ErrTableDropped,
+};
+
+struct ReadFromStreamResult
+{
+    ReadFromStreamError error = ReadFromStreamError::Ok;
+    std::string extra_msg;
+};
+
+static inline std::tuple<ReadFromStreamResult, PrehandleResult> readFromStream(
+    RegionPtr new_region,
+    std::shared_ptr<std::atomic_bool> prehandle_task,
+    DM::FileConvertJobType job_type,
+    std::shared_ptr<StorageDeltaMerge> storage,
+    std::shared_ptr<DM::SSTFilesToBlockInputStream> sst_stream,
+    const DM::SSTFilesToBlockInputStreamOpts & opts,
+    TMTContext & tmt)
```

Comment:
```suggestion
static inline std::tuple<ReadFromStreamResult, PrehandleResult> readFromStream(
    const RegionPtr & new_region,
    const std::shared_ptr<std::atomic_bool> & prehandle_task,
    DM::FileConvertJobType job_type,
    const std::shared_ptr<StorageDeltaMerge> & storage,
    const std::shared_ptr<DM::SSTFilesToBlockInputStream> & sst_stream,
    const DM::SSTFilesToBlockInputStreamOpts & opts,
    TMTContext & tmt)
```

Reply:
And maybe `executeTransform` is a better name for this function?

Reply:
good idea